action/protocol/poll/staking_committee.go

```diff
@@ -67,18 +78,92 @@ func NewStakingCommittee(
+		candidatesByHeight: candidatesByHeight,
+		electionCommittee:  ec,
+		governanceStaking:  gs,
+		nativeStaking:      ns,
+		getEpochHeight:     getEpochHeight,
+		getEpochNum:        getEpochNum,
+		rp:                 rp,
+		scoreThreshold:     scoreThreshold,
 	}, nil
 }
 
-func (sc *stakingCommittee) Initialize(ctx context.Context, sm protocol.StateManager) error {
-	return sc.governanceStaking.Initialize(ctx, sm)
+func (sc *stakingCommittee) CreateGenesisStates(ctx context.Context, sm protocol.StateManager) error {
+	raCtx := protocol.MustGetRunActionsCtx(ctx)
+	if raCtx.BlockHeight != 0 {
+		return errors.Errorf("Cannot create genesis state for height %d", raCtx.BlockHeight)
+	}
+	if gsc, ok := sc.governanceStaking.(protocol.GenesisStateCreator); ok {
+		if err := gsc.CreateGenesisStates(ctx, sm); err != nil {
+			return err
+		}
+	}
+	raCtx.Producer, _ = address.FromString(address.ZeroAddress)
+	raCtx.Caller, _ = address.FromString(nativeStakingContractCreator)
+	raCtx.GasLimit = raCtx.Genesis.BlockGasLimit
+	bytes, err := hexutil.Decode(raCtx.Genesis.NativeStakingContractCode)
+	if err != nil {
+		return err
+	}
+	execution, err := action.NewExecution(
+		"",
+		nativeStakingContractNonce,
+		big.NewInt(0),
+		raCtx.Genesis.BlockGasLimit,
+		big.NewInt(0),
+		bytes,
+	)
+	if err != nil {
+		return err
+	}
+	_, receipt, err := evm.ExecuteContract(
+		protocol.WithRunActionsCtx(ctx, raCtx),
+		sm,
+		execution,
+		func(height uint64) (hash.Hash256, error) {
+			return hash.ZeroHash256, nil
+		},
+	)
+	if err != nil {
+		return err
+	}
+	if receipt.Status != uint64(iotextypes.ReceiptStatus_Success) {
+		return errors.Errorf("error when deploying native staking contract, status=%d", receipt.Status)
+	}
+	sc.SetNativeStakingContract(receipt.ContractAddress)
+	log.L().Info("Deployed native staking contract", zap.String("address", receipt.ContractAddress))
+
+	return nil
+}
+
+func (sc *stakingCommittee) Start(ctx context.Context) error {
+	raCtx := protocol.MustGetRunActionsCtx(ctx)
+	if raCtx.Genesis.NativeStakingContractAddress == "" && raCtx.Genesis.NativeStakingContractCode != "" {
+		caller, _ := address.FromString(nativeStakingContractCreator)
+		ethAddr := crypto.CreateAddress(common.BytesToAddress(caller.Bytes()), nativeStakingContractNonce)
+		iotxAddr, _ := address.FromBytes(ethAddr.Bytes())
+		sc.SetNativeStakingContract(iotxAddr.String())
+		log.L().Info("Loaded native staking contract", zap.String("address", iotxAddr.String()))
+	}
+
+	return nil
+}
+
+func (sc *stakingCommittee) CreatePreStates(ctx context.Context, sm protocol.StateManager) error {
+	if psc, ok := sc.governanceStaking.(protocol.PreStatesCreator); ok {
+		return psc.CreatePreStates(ctx, sm)
+	}
+
+	return nil
+}
+
+func (sc *stakingCommittee) CreatePostSystemActions(ctx context.Context) ([]action.Envelope, error) {
+	if psac, ok := sc.governanceStaking.(protocol.PostSystemActionsCreator); ok {
+		return psac.CreatePostSystemActions(ctx)
+	}
+
+	return nil, nil
```

Comment:
using governanceStaking.CreatePostSystemActions will not call stakingCommittee's DelegatesByHeight, which means it will not merge delegates from eth and native. is that right?